pb-crm-api/src/db/entities/contact.entity.ts

```diff
@@ -1,42 +1,38 @@
-import {
-    Column,
-    CreateDateColumn,
-    Entity,
-    OneToMany,
-    PrimaryGeneratedColumn,
-    UpdateDateColumn,
-} from 'typeorm';
+import {Column, CreateDateColumn, Entity, OneToMany, PrimaryGeneratedColumn, UpdateDateColumn,} from 'typeorm';
+import {IsDefined, IsOptional, Length, Matches, Validate, ValidateNested} from 'class-validator';
+import {Type} from "class-transformer";
+import {ContactModel} from "@hiddentemple/api-interfaces";
 import {EmailEntity} from "./email.entity";
 import {PhoneEntity} from "./phone.entity";
-import {ContactModel} from '@hiddentemple/api-interfaces';
-import { MinLength, MaxLength, IsOptional, IsAlpha, ValidateNested } from 'class-validator';
-import {Type} from "class-transformer";
+import {HasPrimary} from "../../core/validation/has-primary.constraint";
 
+export const NameRegex = /^[a-zA-z-]+$/
 
 @Entity("contacts")
 export class ContactEntity implements ContactModel {
 
     @PrimaryGeneratedColumn('uuid')
     id: string
 
-    @IsAlpha()
-    @Column('varchar', { nullable: false})
-    @MinLength(1)
+    @Column({ type: "character varying", length: 50 })
+    @Matches(NameRegex, {message: "firstName must contain only alphabetic characters and '-'"})
+    @Length(2, 50)
+    @IsDefined()
     firstName: string;
 
-    @IsAlpha()
-    @Column('varchar', { nullable: false})
-    @MinLength(1)
+    @Column({ type: "character varying", length: 50 })
+    @Matches(NameRegex, {message: "lastName must contain only alphabetic characters and '-'"})
```

Comment:
Can't a name have `.` also? e.g. `A.J.`

Reply:
Great point. We should ask the client about any others they can think of. Can you think of any others?

Reply:
Would an apostrophe be an issue as well? i.e. De'wayne or D'Artagnan